client/state/ui/notices/reducer.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+import { filter } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import {
+	NEW_NOTICE,
+	REMOVE_NOTICE
+} from 'state/action-types';
+
+/**
+ * Tracks all known site objects, indexed by site ID.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export function items( state = [], action ) {
```

Comment:
I don't expect to have that many notices shown at once for it to really make a difference.
And there would be an additional effort to keep them sorted chronologically while displaying. 
> The for...in statement iterates over the enumerable properties of an object, in arbitrary order